src/cargo_shim/mod.rs

```diff
@@ -528,6 +528,21 @@ pub enum MessageFormat {
     Json
 }
 
+impl FromStr for MessageFormat {
+    type Err = super::Error;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        match s {
+            "human" => Ok(MessageFormat::Human),
+            "json" => Ok(MessageFormat::Json),
+            _ => Err(super::Error::ConfigurationError(format!(
+                "{} is not a valid message format. Possible values are `human` and `json`.",
+                s
+            ))),
+        }
+    }
+}
+
```

Comment:
*I think* these `FromStr` impls may be unnecessary; won't this work instead?

```rust
#[derive(Copy, Clone, PartialEq, Debug, StructOpt)]
pub enum MessageFormat {
    #[structopt(name = "human")]
    Human,
    #[structopt(name = "json")]
    Json
}
```

Reply:
`derive(StructOpt)` on an enum tries to use the variants as subcommands - it looks like this functionality is not well-supported ([ref 1](https://github.com/TeXitoi/structopt/issues/106)) ([ref 2](https://github.com/TeXitoi/structopt/issues/42)) - basically, it boils down to some kind of parsing boilerplate.

If you don't want to expose a `FromStr` implementation on these types, I can replace it with a method or standalone parsing fn with the same signature. Would you like that solution more?

Reply:
Hmm... well, I guess then the `FromStr` can stay.